compiler/src/dmd/clone.d

```diff
@@ -1648,14 +1650,17 @@ bool needCopyCtor(StructDeclaration sd, out bool hasCpCtor)
             return 0;
         }
 
-        if (isRvalueConstructor(sd, ctorDecl))
+        if (ctorDecl.isMoveCtor)
             rvalueCtor = ctorDecl;
         return 0;
     });
 
+    if (rvalueCtor)
+        hasMoveCtor = true;
+
     if (cpCtor)
     {
-        if (rvalueCtor)
+        if (0 && rvalueCtor)
```

Comment:
is there any reason to leave this dead branch here?

Reply:
I left it as a marker of the bifurcation point for rvalue constructors. I'll probably yank it soon.